components/List/examples/MDCSelfSizingStereoCellExample.m

```diff
@@ -122,21 +126,50 @@ - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView
   return cell;
 }
 
-- (NSString *)generateRandomString {
-  NSInteger numberOfWords = 0 + arc4random() % (25 - 0);
-  NSMutableArray *wordArray = [[NSMutableArray alloc] initWithCapacity:numberOfWords];
-  for (NSInteger i = 0; i < numberOfWords; i++) {
-    NSInteger lengthOfWord = 0 + arc4random() % (10 - 0);
-    NSMutableArray *letterArray = [[NSMutableArray alloc] initWithCapacity:lengthOfWord];
-    for (NSInteger j = 0; j < lengthOfWord; j++) {
-      int asciiCode = 97 + arc4random() % (122 - 97);
-      NSString *characterString = [NSString stringWithFormat:@"%c", asciiCode];
-      [letterArray addObject:characterString];
+- (NSTextAlignment)textAlignmentForText:(NSString *)text {
+  if (text.length > 0) {
+    if (text.mdf_calculatedLanguageDirection == NSLocaleLanguageDirectionLeftToRight) {
```

Comment:
There may be some performance impact from using mdf_calculatedLanguageDirection in cellForIndexPath, but if everything looks okay, LGTM.

Reply:
Was the performance impact measured in scrolling? Can we still hit 60 fps on an iPhone 5c/iPhone 4S?

Reply:
I did a quick-and-dirty test on an iPhone 5c/iOS 10.3.3 and found that for 3.8s worth of casual scrolling (over an 8-10 second period, depending on trace), it added about 25 ms worth of overhead (~0.6% increase).

Reply:
Interesting! Are we okay with this?

Reply:
I dropped the performance to about half by caching the result in #5506